buildtest/modules/spider.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+import subprocess
+
+from buildtest.tools.defaults import BUILDTEST_SPIDER_FILE, BUILDTEST_CONFIG_FILE
+from buildtest.tools.config import load_configuration
+
+
+class Spider:
+    """Class declaration of Spider class"""
+
+    def __init__(self, tree=None):
+        """Initialize method for Spider class.
+
+        :param tree: User can specify a module tree to query from spider.
+        :type tree: str
+        """
+
+        #  if user specifies a tree, then run spider command otherwise read from configuration file.
+        if tree:
+            self.tree = tree
+            spider_cmd = f"{os.getenv('LMOD_DIR')}/spider -o spider-json {self.tree}"
+
+            out = subprocess.check_output(spider_cmd, shell=True).decode("utf-8")
+            self.spider_content = json.loads(out)
+
+        else:
+            with open(BUILDTEST_SPIDER_FILE, "r") as fd:
+                self.spider_content = json.load(fd)
+
+            content = load_configuration(BUILDTEST_CONFIG_FILE)
```

Comment:
Is this guaranteed to load something, and is the something guaranteed to have `BUILDTEST_MODULEPATH`? The load_configuration should likely exit cleanly if anything weird happens and then the self.tree should get like:

```
self.tree = content.get("BUILDTEST_MODULEPATH", [])
```